pkg/setup/initialize.go

```diff
@@ -31,8 +31,14 @@ import (
 
 // Initialize creates the required secrets and configmaps for the control plane to start
 func Initialize(ctx context.Context, options *config.VirtualClusterConfig) error {
+	// migrate k3s to k8s if needed
+	err := k8s.MigrateK3sToK8s(ctx, options.ControlPlaneClient, options.ControlPlaneNamespace, options)
```

Comment:
should we run this if embedded etcd is enabled? I think we already have a data migration logic for sqlite - > embedded etcd in the pro.

Reply:
yeah this is about migrating the distro not the backing store, so this wouldn't cause any issues